pkg/collector/reconcile/service_test.go

```diff
@@ -0,0 +1,231 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconcile
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/open-telemetry/opentelemetry-operator/api/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/internal/config"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector"
+)
+
+func TestExtractPortNumbersAndNames(t *testing.T) {
+	t.Run("should return extracted port names and numbers", func(t *testing.T) {
```

Comment:
Sorry, should have caught this before on the other PR as well, but why are you wrapping the test in a test?

Reply:
Just give a proper name to the test - t.Run gives a way to describe the test properly using name field and to group few tests which belong together.

I understand that for function with only  one test it might look odd. I can remove it

Reply:
That's fine, not sure I quite like the fact that we now have two patterns in the test suite, but we can make this the standard for the reconciliation tests.